### DISCUSSION TITLE
post/producer-consumer-pattern-tpl

### DISCUSSION THREAD

**Discussion by giscus** [markheath/soundcode-blog-comments#20] [Announcements]
# post/producer-consumer-pattern-tpl

Mark Heath's Development Blog

https://markheath.net/post/producer-consumer-pattern-tpl

**Comment by isensible** [comment]
I've recently used TPL Dataflow to consume messages from Azure EventHub. 

The EventHub client is not as richly featured as the ServiceBus client. EventHub exists for a different reason, of course. TPL Dataflow allowed me to treat EventHub more like ServiceBus, giving me precise control of parallel event processing.

In the end, the customer wised up and switched to ServiceBus, which meant I no longer needed my TPL Dataflow blocks.

**Comment by alexdresko** [comment]
I've been using System.Threading.Channels with great success for this kind of thing.

**Reply by markheath** [reply]
Yes, I want to give that a try as well soon.
